arrow-select/src/concat.rs

```diff
@@ -611,7 +612,7 @@ mod tests {
             ],
         )
         .unwrap();
-        let new_batch = concat_batches(&schema, &[batch1, batch2]).unwrap();
+        let new_batch = concat_batches(&schema, [&batch1, &batch2]).unwrap();
         assert_eq!(new_batch.schema().as_ref(), schema.as_ref());
         assert_eq!(2, new_batch.num_columns());
         assert_eq!(4, new_batch.num_rows());
```

Comment:
I think it would be good to keep one example of the previous API too if we are going to claim this API is no different. Perhaps something like:


```suggestion
        assert_eq!(new_batch.schema().as_ref(), schema.as_ref());
        assert_eq!(2, new_batch.num_columns());
        assert_eq!(4, new_batch.num_rows());

        // ensure previous API compiles as well
        let new_batch_owned = concat_batches(&schema, &[batch1, batch2]).unwrap();
        assert_eq!(new_batch, new_batch_owned);
```

?